systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java

```diff
@@ -0,0 +1,389 @@
+import io.strimzi.api.kafka.model.KafkaUserScramSha512ClientAuthentication;
+import io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListenerBuilder;
+import io.strimzi.api.kafka.model.listener.arraylistener.KafkaListenerType;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.annotations.ParallelTest;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaAdminClient;
+import io.strimzi.systemtest.resources.crd.kafkaclients.AdminClientOperations;
+import io.strimzi.systemtest.resources.operator.SetupClusterOperator;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.List;
+import java.util.Random;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@Tag(REGRESSION)
+@Tag(INTERNAL_CLIENTS_USED)
+public class ThrottlingQuotaST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(ThrottlingQuotaST.class);
+    static final String NAMESPACE = "throttling-quota-cluster-test";
+    static final String CLUSTER_NAME = "quota-cluster";
+    final String kafkaUsernamePrefix = "test-quota-user";
+    final String classTopicPrefix = "quota-topic-test";
+
+    /**
+     * Test checks for throttling quotas set for user
+     * on creation & deletion of topics and create partition operations.
+     */
+    @ParallelTest
+    void testThrottlingQuotasCreateTopic(ExtensionContext extensionContext) {
+        final String kafkaUsername = kafkaUsernamePrefix + "-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String createAdminName = "create-admin-client" + "-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String topicNamePrefix = classTopicPrefix + "-create";
```

Comment:
Is it intention to do it this way? Because you should use maps with usernames, topic names etc. Something like this:
```
mapWithTestUsers.get(extensionContext.getDisplayName());
```

Reply:
I don't know. Does it always generate new username? It seems to me, that it only generates one KU.

Reply:
For topic name, I don't need randomization, as each test uses unique topic name (with generated suffix numbers)

Reply:
You can get client name from the map too. And `mapWithTestUsers.get(extensionContext.getDisplayName());` will give you specific username for each test case. If I'm looking right, you are using just one user per test. So using the code above will be better FMPOV.